pkg/reader/namespace/namespace.go

```diff
@@ -180,30 +207,49 @@ func GetMsgNamespaces(ns processapi.MsgNamespaces) *tetragon.Namespaces {
 		retVal.TimeForChildren = nil
 	}
 
-	return retVal
+	return retVal, nil
 }
 
-func GetHostNamespace() *tetragon.Namespaces {
-	if hostNamespace == nil {
-		hostNamespace = &tetragon.Namespaces{
-			Uts:             createHostNs("uts"),
-			Ipc:             createHostNs("ipc"),
-			Mnt:             createHostNs("mnt"),
-			Pid:             createHostNs("pid"),
-			PidForChildren:  createHostNs("pid_for_children"),
-			Net:             createHostNs("net"),
-			Time:            createHostNs("time"),
-			TimeForChildren: createHostNs("time_for_children"),
-			Cgroup:          createHostNs("cgroup"),
-			User:            createHostNs("user"),
+func InitHostNamespace() (*tetragon.Namespaces, error) {
+	if hostNamespaces != nil {
+		return hostNamespaces, nil
+	}
+
+	kernelVer, _, _ := kernels.GetKernelVersion(option.Config.KernelVersion, option.Config.ProcFS)
+	// time and time_for_children namespaces introduced in kernel 5.6
+	TimeNsSupport = (int64(kernelVer) >= kernels.KernelStringToNumeric("5.6.0"))
```

Comment:
Can we check if the kernel supports time namespace by checking if `/proc/1/ns/time` exists? I believe that this is a good idea not to rely on the kernel version. For example, the 4.18 redhat kernel may have backported time namespace.

(I know that we did that before, maybe it is a good time to fix that.)